synapse/util/logcontext.py

```diff
@@ -359,6 +363,25 @@ def g(*args, **kwargs):
     return g
 
 
+@defer.inlineCallbacks
+def make_deferred_yieldable(deferred):
```

Comment:
errrrrrrrr, !invoke naming-committee,  (I can't think of anything better either)